digdag-standards/src/main/java/io/digdag/standards/operator/td/TdDdlOperatorFactory.java

```diff
@@ -110,25 +117,71 @@ public TaskResult runTask()
                     op.withDatabase(t.getDatabase().or(op.getDatabase())).ensureTableCreated(t.getTable());
                 });
             }
+            for (RenameTableConfig r : renameTableList) {
+                TableParam from = r.getFromTable();
+                String to = r.getToTable();
+                if (to.contains(".")) {
+                    // renaming a table across databases is not supported by the td's rest api itself
+                    throw new ConfigException("'to' option of rename_tables can't include database name");
+                }
+                operations.add(op -> {
+                    logger.info("Renaming TD table {}.{} -> {}", op.getDatabase(), from, to);
+                    op.withDatabase(from.getDatabase().or(op.getDatabase())).ensureExistentTableRenamed(from.getTable(), to);
+                });
+            }
 
             try (TDOperator op = TDOperator.fromConfig(env, params, context.getSecrets().getSecrets("td"))) {
+                // make sure that all "from" tables exist so that ignoring 404 Not Found in
+                // op.ensureExistentTableRenamed is valid.
+                if (!renameTableList.isEmpty()) {
+                    int checkOperation = state.params().get("rename_check_operation", int.class, 0);
+                    for (int i = checkOperation; i < renameTableList.size(); i++) {
+                        state.params().set("rename_check_operation", i);
+
+                        RenameTableConfig r = renameTableList.get(i);
+                        TableParam from = r.getFromTable();
+                        String database = from.getDatabase().or(op.getDatabase());
+
+                        boolean exists = pollingRetryExecutor(state, "rename_check_retry")
+                                .retryUnless(TDOperator::isDeterministicClientException)
+                                .withRetryInterval(retryInterval)
+                                .withErrorMessage("Failed check existance of table %s.%s", database, from.getTable())
+                                .runOnce(boolean.class, s -> {
```

Comment:
`s/runOnce/run/`

Otherwise the `"rename_check_retry"` will only be executed once. I.e., only the existence of the first table will be checked.

I guess the fact that the tests for this PR are passing means that the test coverage is insufficient.